### ISSUE TITLE
AttributeError: 'Name' object has no attribute 'infered' with python3.7

### ISSUE THREAD

**Issue by pavelrevak** [PyCQA/pylint#2238]
### Steps to reproduce
1. create file with this code (yes, it has no sense, because function b() does not exists, but it should not stop on error)
```python
def a():
    b()
    return None
```
or also this code:
```python
def a():
    print()
    return None
```
it has no effect what function is on line 2.

2. run pylint without any params on this file
```bash
pylint test.py
```

### Current behavior
I see this error:
```
pylint test.py 
************* Module test
test.py:1:0: C0111: Missing module docstring (missing-docstring)
test.py:1:0: C0103: Function name "a" doesn't conform to snake_case naming style (invalid-name)
test.py:1:0: C0111: Missing function docstring (missing-docstring)
test.py:2:4: E0602: Undefined variable 'b' (undefined-variable)
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/bin/pylint", line 11, in <module>
    sys.exit(run_pylint())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/__init__.py", line 18, in run_pylint
    Run(sys.argv[1:])
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/lint.py", line 1371, in __init__
    linter.check(args)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/lint.py", line 783, in check
    self._do_check(files_or_modules)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/lint.py", line 916, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/lint.py", line 996, in check_astroid_module
    walker.walk(ast_node)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/utils.py", line 1140, in walk
    self.walk(child)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/utils.py", line 1142, in walk
    cb(astroid)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/checkers/refactoring.py", line 384, in leave_functiondef
    self._check_consistent_returns(node)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/checkers/refactoring.py", line 706, in _check_consistent_returns
    and self._is_node_return_ended(node)):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/checkers/refactoring.py", line 767, in _is_node_return_ended
    return any(self._is_node_return_ended(_child) for _child in node.get_children()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/checkers/refactoring.py", line 768, in <genexpr>
    if not isinstance(_child, astroid.ExceptHandler))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/checkers/refactoring.py", line 767, in _is_node_return_ended
    return any(self._is_node_return_ended(_child) for _child in node.get_children()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/checkers/refactoring.py", line 768, in <genexpr>
    if not isinstance(_child, astroid.ExceptHandler))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pylint/checkers/refactoring.py", line 725, in _is_node_return_ended
    funcdef_node = node.func.infered()[0]
AttributeError: 'Name' object has no attribute 'infered'
```

### Expected behavior
 No AttributeError

### pylint --version output
```
pylint 2.0.0.dev1
astroid 2.0.0.dev3
Python 3.7.0 (v3.7.0:1bf9cc5093, Jun 26 2018, 23:26:24) 
[Clang 6.0 (clang-600.0.57)]
```
installed today with:
```bash
pip3 install pylint astroid --pre -U
```

**Comment by adamtheturtle** [comment]
FWIW the attribute is spelled `inferred`.

**Comment by adamtheturtle** [comment]
This appears to be fixed in https://github.com/PyCQA/pylint/commit/a9aef1321b599e2cd4110b2c72e7f62f55c084aa meaning that a release should fix this issue.

**Comment by PCManticore** [comment]
This is indeed fixed with the linked commit. We'll do another dev release this week, followed by the actual 2.0 next week.
